src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -369,6 +377,15 @@ public ReplicationStatus getStatus() {
      */
     @Override
     public void changed(ChangeEvent event) {
+
+        // forget cached IDs (Should be executed in workExecutor)
+        db.runAsync(new AsyncTask() {
+            @Override
+            public void run(Database database) {
+                pendingDocIDs = null;
```

Comment:
If SYNC mode, whatever original thread is.
If ASYNC mode, it is called from workExecutor (single thread).

However, `changeListenerNotifyStyle` is always `ChangeListenerNotifyStyle.ASYNC`. 

```java
f (changeListenerNotifyStyle == ChangeListenerNotifyStyle.SYNC) {
            for (ChangeListener changeListener : changeListeners) {
                try {
                    changeListener.changed(changeEvent);
                } catch (Exception e) {
                    e.printStackTrace();
                    Log.e(Log.TAG_SYNC, "Exception notifying replication listener: %s", e);
                }
            }
        } else {
            // ASYNC
            synchronized (workExecutor) {
                if (!workExecutor.isShutdown()) {
                    workExecutor.submit(new Runnable() {
                        @Override
                        public void run() {
                            try {
                                for (ChangeListener changeListener : changeListeners) {
                                    changeListener.changed(changeEvent);
                                }
                            } catch (Exception e) {
                                Log.e(Log.TAG_SYNC, "Exception notifying replication listener: %s", e);
                                throw new RuntimeException(e);
                            }
                        }
                    });
                }
            }
        }
```